manifests/check.pp

```diff
@@ -32,22 +42,35 @@
 #   Human readable description of the check
 #
 define consul::check(
-  $id         = $title,
-  $ttl        = undef,
-  $http       = undef,
-  $script     = undef,
-  $interval   = undef,
-  $service_id = undef,
-  $timeout    = undef,
-  $notes      = undef,
+  $id            = $title,
+  $check_ensure  = present,
+  $ttl           = undef,
+  $http          = undef,
+  $script        = undef,
+  $script_source = undef,
+  $script_path   = undef,
+  $interval      = undef,
+  $service_id    = undef,
+  $timeout       = undef,
+  $notes         = undef,
 ) {
   include consul
 
+  # Deploy script if necessary
+  if $script_source {
+    file { "${id}":
```

Comment:
shouldn't this just be $script and then we don't need the $script_path variable?